### PR TITLE
apache-arrow: Bugfix for different Python installations

### PR DESCRIPTION
**Pull request by essandess**
* Fixes a conflict between conflicting apache-arrow and Python libraries

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/apache-arrow/Portfile [hunk 1]
```diff
@@ -45,13 +47,41 @@ depends_build-append \
...
         ui_error "${name} @${version} requires macOS 10.13 or later."
-        return -code error "incompatible macOS version"
+        return -code error "incompatible OS X version"
+    }
+}
+
+pre-configure {
+    xinstall -d ${workpath}/bin
+    foreach f [glob -nocomplain ${prefix}/bin/ll*-${llvm_version}] {
+        ln -s ${f} ${workpath}/bin/[strsed [file tail ${f}] "s|-${llvm_version}$||"]
```
Comment (cjones051073) [inline devel/apache-arrow/Portfile:84]: I dont see why this complication is necessary, when you could just directly add the path to the LLVM libexec install area to PATH, as was done in https://github.com/macports/macports-ports/commit/32307f47e04ab9abb61d3d3fa610bfd3440d0a5b#diff-48798ca325280ad7ee3005fd6db459c1b47015fc2397f5809502e4bf38066501 ?

**Review by cjones051073 — COMMENTED** [review]
The diffs here are extensive and it is difficult to see what conflict exactly you are fixing ?

File: devel/apache-arrow/Portfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ PortGroup           cmake 1.1
 PortGroup           github 1.0
 
 github.setup        apache arrow 3.0.0 apache-arrow-
-revision            6
+revision            4
```
Comment (cjones051073) [inline devel/apache-arrow/Portfile:8]: You are decreasing the revision here which is not allowed.

this, and a bunch if other changes in the PR tells me you have not properly based your changes on the current version in master. Please go back fix this to make sure the diff here is only the minimal changes need to fix whatever the conflict is you refer to.

File: devel/apache-arrow/Portfile [hunk 3]
```diff
@@ -24,17 +24,19 @@ checksums           rmd160  e2dbd95f5c9b5739a7351dd920b40509c4e968ca \
                     sha256  94cca341bc595881e828bde5a9e08ee58287ade9e5bcd070cd6d2a0cd8f0bb4a \
                     size    7518019
 
-compiler.cxx_standard 2017
+compiler.cxx_standard 2011
```
Comment (cjones051073) [inline devel/apache-arrow/Portfile:27]: Please do not make this change, as 2017 was found to be required for older systems.

File: devel/apache-arrow/Portfile [hunk 4]
```diff
@@ -24,17 +24,19 @@ checksums           rmd160  e2dbd95f5c9b5739a7351dd920b40509c4e968ca \
...
 
 # Build instructions: https://github.com/apache/arrow/blob/master/docs/source/developers/python.rst#build-and-test
+cmake.generator     Ninja
 cmake.source_dir    ${worksrcpath}/cpp
 cmake.build_type    Release
+set cmake_build_type \
+                    [string tolower ${cmake.build_type}]
 
-set llvm_version    11
+set llvm_version    10
```
Comment (cjones051073) [inline devel/apache-arrow/Portfile:38]: Please do not make this change

File: devel/apache-arrow/Portfile [hunk 5]
```diff
@@ -208,56 +198,119 @@ if {[string match "py*" ${subport}]} {
...
-                    --build-type=${arrow_build_type} \
-                    --bundle-arrow-cpp bdist_wheel
+    build.cmd       ${python.bin} setup.py --no-user-cfg
+    build.target    bdist_wheel
+    build.post_args
 
     destroot.env-append \
-                    "PATH=${python.prefix}/bin:${llvm_bin}:$env(PATH)" \
+                    "${python.prefix}/bin:${workpath}/bin:$env(PATH)" \
                     CCACHE_DIR=${workpath}/.ccache
```
Comment (cjones051073) [inline devel/apache-arrow/Portfile:293]: Surely the PATH= is correct here ?

**Review by cjones051073 — CHANGES_REQUESTED** [review]

File: devel/apache-arrow/Portfile (hunk 1 above)
Comment (essandess) [reply devel/apache-arrow/Portfile:84]: > The diffs here are extensive and it is difficult to see what conflict exactly you are fixing ?

A refactor is necessary to address this issue:
```bash
$ python3.8 -c 'import pyarrow.lib as _lib'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pyarrow/__init__.py", line 63, in <module>
    import pyarrow.lib as _lib
ImportError: dlopen(/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pyarrow/lib.cpython-38-darwin.so, 2): Symbol not found: __Py_FatalErrorFunc
  Referenced from: /opt/local/lib/libarrow_python.300.dylib
  Expected in: flat namespace
 in /opt/local/lib/libarrow_python.300.dylib
```

The `py*-pyarrow` ports must build separately from the `apache-arrow` port.

**Comment by essandess** [comment]
> The diffs here are extensive and it is difficult to see what conflict exactly you are fixing ?

A refactor is necessary to address this issue:
```bash
$ python3.8 -c 'import pyarrow.lib as _lib'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pyarrow/__init__.py", line 63, in <module>
    import pyarrow.lib as _lib
ImportError: dlopen(/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pyarrow/lib.cpython-38-darwin.so, 2): Symbol not found: __Py_FatalErrorFunc
  Referenced from: /opt/local/lib/libarrow_python.300.dylib
  Expected in: flat namespace
 in /opt/local/lib/libarrow_python.300.dylib
```

The `py*-pyarrow` ports must build separately from the `apache-arrow` port.

File: devel/apache-arrow/Portfile (hunk 1 above)
Comment (essandess) [reply devel/apache-arrow/Portfile]: Thanks for doing this the right way. I started working on this refactor before your improvements were added and obviously missed these. The latest push is up-to-date.

File: devel/apache-arrow/Portfile (hunk 2 above)
Comment (essandess) [reply devel/apache-arrow/Portfile]: Yep, I started working on this refactor before your improvements were added and obviously missed these. The latest push is up-to-date.

File: devel/apache-arrow/Portfile (hunk 3 above)
Comment (essandess) [reply devel/apache-arrow/Portfile]: Fixed thanks.

File: devel/apache-arrow/Portfile (hunk 4 above)
Comment (essandess) [reply devel/apache-arrow/Portfile]: Fixed thanks.

File: devel/apache-arrow/Portfile (hunk 5 above)
Comment (essandess) [reply devel/apache-arrow/Portfile:293]: Correct, and unnecessary. The `cmake` options take care of this. Fixed by deletion.

**Review by cjones051073 — APPROVED** [review]

**Pull request merged**